src/wp-includes/block-template-utils.php

```diff
@@ -1468,6 +1468,22 @@ function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
 	$template = $request['id'] ? get_block_template( $request['id'], $post_type ) : null;
```

Comment:
This call to `get_block_template` results in an extra invocation of the `hooked_block_types` filter.

As @tjcafferkey has mentioned in DM to me, this is visible in the newly added unit test, where the mock filter is run by the Block Hooks logic which is attempting insertion of hooked blocks into the mock template's content (which is just a `Content` string literal, i.e. no blocks).